src/main/java/com/sourcegraph/cody/chat/BlinkingCursorComponent.java

```diff
@@ -0,0 +1,41 @@
+package com.sourcegraph.cody.chat;
+
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.Graphics;
+import javax.swing.JPanel;
+import javax.swing.Timer;
+
+public class BlinkingCursorComponent extends JPanel {
+  private boolean showCursor;
+  private Timer timer;
+
+  public static BlinkingCursorComponent instance = new BlinkingCursorComponent();
+
+  private BlinkingCursorComponent() {
+    this.showCursor = true;
+    this.timer =
+        new Timer(
+            500,
+            e -> {
+              showCursor = !showCursor;
+              repaint();
+            });
+    this.timer.start();
```

Comment:
The tests fail. In CI the info is very brief. In a local run (`./gradlew test`) I see the following error:

```
junit.framework.AssertionFailedError: Not disposed javax.swing.Timer: Timer (listeners: [com.sourcegraph.cody.chat.BlinkingCursorComponent$$Lambda$908/0x0000000800a00840@160f9d46]) (delayed for 379ms); queue:TimerQueue (javax.swing.Timer@970746c)
	at com.intellij.testFramework.TestApplicationManagerKt.checkJavaSwingTimersAreDisposed(TestApplicationManager.kt:355)
	at com.intellij.testFramework.TestApplicationManagerKt.tearDownProjectAndApp(TestApplicationManager.kt:207)
	at com.intellij.testFramework.fixtures.impl.LightIdeaTestFixtureImpl.lambda$tearDown$3(LightIdeaTestFixtureImpl.java:79)
	at com.intellij.testFramework.RunAllKt.doRun(runAll.kt:50)
	at com.intellij.testFramework.RunAllKt.access$doRun(runAll.kt:1)
	at com.intellij.testFramework.RunAll.run(runAll.kt:17)
	at com.intellij.testFramework.fixtures.impl.LightIdeaTestFixtureImpl.tearDown(LightIdeaTestFixtureImpl.java:105)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$tearDown$33(CodeInsightTestFixtureImpl.java:1326)
	at com.intellij.testFramework.EdtTestUtil.runInEdtAndWait(EdtTestUtil.java:33)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.lambda$tearDown$34(CodeInsightTestFixtureImpl.java:1326)
	at com.intellij.testFramework.RunAllKt.doRun(runAll.kt:50)
	at com.intellij.testFramework.RunAllKt.doRun$default(runAll.kt:46)
	at com.intellij.testFramework.RunAll$Companion.runAll(runAll.kt:27)
	at com.intellij.testFramework.RunAll.runAll(runAll.kt)
	at com.intellij.testFramework.fixtures.impl.CodeInsightTestFixtureImpl.tearDown(CodeInsightTestFixtureImpl.java:1284)
	at com.intellij.testFramework.fixtures.BasePlatformTestCase.tearDown(BasePlatformTestCase.java:62)
	at com.intellij.testFramework.UsefulTestCase.invokeTearDown(UsefulTestCase.java:483)
	at com.intellij.testFramework.UsefulTestCase.defaultRunBare(UsefulTestCase.java:471)
	at com.intellij.testFramework.UsefulTestCase.lambda$runBare$12(UsefulTestCase.java:541)
	at com.intellij.testFramework.EdtTestUtil.lambda$runInEdtAndWait$1(EdtTestUtil.java:40)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:303)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:393)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:207)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:128)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:117)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:105)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:92)
```

Reply:
Before seeing your comment, I was going to check if we dispose this timer. I'm surprised our CI caught this, I thought we don't have enough test coverage to catch these kinds of bugs.